content/blog/survival-analysis-parsnip-adjacent/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+
+### Model fitting
+
+For fitting a model, parsnip typically supports a formula and a matrix interface, regardless of the interface of the engine. For censored regression models, we are currently only supporting a formula interface. Most engines already support this type of interface, and a formula allows us to specify the outcome via the standard `survival::Surv()` function, tying together event time and status. 
+
+
+A formula also let's us specify stratification for a proportional hazards model. For the `survival::coxph()` function, which powers the `survival` engine for `proportional_hazards()`, this may look like 
+
+<div align="center">
+  `Surv(time, event) ~ x + z + strata(s)`.
+</div>
+
+
+The glmnet package, which powers the `glmnet` engine, does not have a formula interface. It requires us to specify the strata by stratifying the response via `stratifySurv(y, s)`. The response is typically a `Surv` object, so you could end up with a formula like this:
+
+<div align="center">
+  `stratifySurv(Surv(time, event), strata) ~ x`.
+</div>
+
+This works for fitting the model but the syntax is rather convoluted on the left-hand side of the formula. The specification of the stratification is inconsistent with the approach taken by the survival package. Additionally, without further modification, prediction fails because variables on the left-hand side of the formula are treated as response variables by parsnip and not available at prediction time.
+
+We already need to translate from formula interface to matrix interface for this engine, and we like interfaces to be as consistent as possible here in tidymodels so we decided to go with the specification of stratification through a `strata()` term on the right-hand side of the formula for proportional hazards models, regardless of the engine.
+
+```{r strata}
+library(survival)
+
+bladder_train <- bladder[-c(1:3),]
+bladder_test <- bladder[1:3,]
+
+cox_spec <- proportional_hazards(penalty = 0.123) %>%
+    set_engine("glmnet")
+
+f_fit <- fit(cox_spec,
+             Surv(stop, event) ~ rx + size + number + strata(enum),
+             data = bladder_train)
+```
+
+
+### Prediction 
+
+We have introduced several new prediction types: time to event (`time`), linear predictor (`linear_pred`), survival probability (`survival`), and hazard (`hazard`). Some of these types are time-dependent, e.g., the survival probability is the probability to survive beyond a certain time point. In keeping with the tidymodels design principle that each row of the data set you're predicting on should give you exactly one row in the dataset of predictions, `predict(type = "survival", time)` returns a nested tibble if `time` contains multiple time points.
+
+```{r predict}
+f_pred <- predict(f_fit, new_data = bladder_test,
+                  type = "survival", time = c(5, 10, 15, 20))
+
+# 3 rows for 3 observations
+f_pred 
```

Comment:
I would use a bunch of time points, `unnest()` them, and show a ggplot of survival curves for a few samples. This would help demonstrate the utility of what we are doing.

Reply:
+1 for showing how to unnest this structure